### ISSUE TITLE
Warning on building=roof

### ISSUE THREAD

**Issue by alesarrett** [openstreetmap/id-tagging-schema#1432]
### How to reproduce the issue?

When creating a building=roof, iD is raising a warning if a layer=1 is not entered, even if the roof is not intersecting any other feature.
Anyway in the wiki is clear that layer=1 should be used when 2 or more features overlap, to express vertical positioning. Also the wiki page for roof is describing the use of the key layer only when roof is overlapping with something else: https://wiki.openstreetmap.org/wiki/Tag:building%3Droof
So, why this requirement and warning on iD?

### Screenshot(s) or anything else?

![Screenshot_20240920_115017](https://github.com/user-attachments/assets/66bb80cd-a968-4f1d-b8ff-cbaa4d1a02b6)

### Which deployed environments do you see the issue in?

Released version at openstreetmap.org/edit

### Which browsers are you seeing this problem on?

Firefox

**Comment by tyrasd** [comment]
> users may add a layer=1 unnecessarily

This is true, but that extra tag does also not create any problems AFAIK.

The adding of the `layer` tag was introduced because quite a high proportion of roofs are tagged with a `layer` even before iD suggested to add it, see #227:

> Currently, 43% of roofs include layer=*

I would argue that it is in general very likely that `building=roof` objects have some other features [underneath](https://wiki.openstreetmap.org/wiki/Tag:building%3Droof#Examples) them (e.g. the service road of a fuel station, footpaths, vending machines, etc.). Adding `layer=1` by default to roofs helps mappers to streamline the mapping of the whole situation.

<hr>

That said, for existing features, it is indeed not necessary to show the warning if there are no other overlapping features, as in your screenshot. That's fixed now with 26c1ff1
